packages/workbox-precaching/controllers/PrecacheManager.mjs

```diff
@@ -0,0 +1,99 @@
+import {_private} from 'workbox-core';
+import core from 'workbox-core';
+
+import PrecacheEntry from '../models/PrecacheEntry.mjs';
+import showWarningsIfNeeded from '../utils/showWarningsIfNeeded.mjs';
+
+/**
+ * Performs efficient precaching of assets.
+ */
+export default class PrecacheManager {
+  /**
+   * Create a new PrecacheManager Instance
+   */
+  constructor() {
+    this._entriesToCacheMap = new Map();
+    if (process.env.NODE_ENV !== 'prod') {
```

Comment:
Another thing I noticed today when looking at the v3 branch is we're using `'prod'` instead of `'production'`, which I think is more common in the npm community. We should confirm this and stick with whatever is most common.